packages/studio-ui/src/hooks/useInjectUserStyles.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from "react";
+import useStudioStore from "../store/useStudioStore";
+
+export default function useInjectUserStyles(
+  iframeDocument: Document | undefined
+) {
+  const [
+    componentTree,
+    getComponentMetadata,
+    activePage,
+    UUIDToFileMetadata,
+    pageCss,
+  ] = useStudioStore((store) => [
+    store.actions.getComponentTree(),
+    store.actions.getComponentMetadata,
+    store.pages.activePageName,
+    store.fileMetadatas.UUIDToFileMetadata,
+    store.pages.getActivePageState()?.cssImports,
+  ]);
+  useEffect(() => {
+    if (!iframeDocument) {
+      return;
+    }
+    componentTree?.forEach((component) => {
+      const cssImports = getComponentMetadata(component)?.cssImports;
+      cssImports?.forEach((cssFilepath) => {
+        injectStyleIntoIframe(iframeDocument, cssFilepath);
+      });
+    });
+    pageCss?.forEach((cssFilepath) => {
+      injectStyleIntoIframe(iframeDocument, cssFilepath);
+    });
+
+    return () => {
+      clearStylingFromIframe(iframeDocument);
+    };
+  }, [
+    componentTree,
+    getComponentMetadata,
+    iframeDocument,
+    pageCss,
+    activePage,
+    UUIDToFileMetadata,
+  ]);
+}
+
+function clearStylingFromIframe(iframeDocument: Document) {
+  const styleElements = [...iframeDocument.head.getElementsByTagName("style")];
+  styleElements.forEach((element) => {
+    element.remove();
+  });
+}
+
+function injectStyleIntoIframe(iframeDocument: Document, filepath: string) {
+  const originalStyletag = document.querySelector(
+    `[studio-style-filepath='${filepath}']`
+  );
+  if (originalStyletag) {
```

Comment:
would we want to log some kind of warning if the styletag isn't found? what are the scenarios in which that could happen?

Reply:
The only time where that would be the case is if the user adds a completely new CSS import to a page or component.  In this case, the filemetadata will be updated through component HMR, but since we only add styletags to Studio's parent head at the beginning of Studio load, a corresponding styletag for the new CSS file won't be found. I can add a warning 👍